### PR TITLE
Task Track - Create

### PR DESCRIPTION
**Pull request by leonidbogin**
мммк

### CODE REVIEW

File: DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/ITaskTrackService.cs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+﻿using DIMS_Core.BusinessLayer.Models.TaskTrack;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace DIMS_Core.BusinessLayer.Interfaces
+{
+    public interface ITaskTrackService
+    {
+        Task<IEnumerable<VTaskTrackModel>> GetAllForMember(int UserId);
+
```
Comment (SashaDrapun) [inline DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/ITaskTrackService.cs:10]: The same remarks as Konstantin's. Camelcase for method params.

File: DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/IUserTaskService.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using DIMS_Core.BusinessLayer.Models.UserTask;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace DIMS_Core.BusinessLayer.Interfaces
+{
+    public interface IUserTaskService
+    {
+        Task<IEnumerable<UserTaskModel>> GetAllForMember(int UserId);
```
Comment (SashaDrapun) [inline DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/IUserTaskService.cs:9]: Same remark. Camelcase for method params.

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+namespace DIMS_Core.Controllers
+{
+    [Route("task-tracks")]
+    public class TaskTrackController : Controller
+    {
+        private readonly ITaskTrackService taskTracksService;
+        private readonly IUserTaskService userTaskService;
+        private readonly IMapper mapper;
+
+        public TaskTrackController(ITaskTrackService taskTracksService, IUserTaskService userTaskService, IMapper mapper)
```
Comment (SashaDrapun) [inline DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:20]: Split parameters into several lines if the number of parameters passed is more than 2.

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+            this.mapper = mapper;
+        }
+
+        [HttpGet("")]
+        public async Task<IActionResult> Index()
+        {
+            // To Do - Get the id of the current user
+            int userId = 3;
+
+            var searchResult = await taskTracksService.GetAllForMember(UserId: userId);
```
Comment (SashaDrapun) [inline DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:33]: Can't we pass the parameter in the usual way?

**Review by SashaDrapun — COMMENTED** [review]
Hello good job! I found a couple of shortcomings, please fix them, or explain.

File: DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/ITaskTrackService.cs (hunk 1 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/ITaskTrackService.cs]:
```suggestion
        Task<IEnumerable<VUserTrackModel>> GetAllByUserId(int userId);

File: DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/IUserTaskService.cs (hunk 2 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/IUserTaskService.cs:9]:
```suggestion

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs (hunk 3 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:20]: private readonly IUserTaskService userTaskService;
        private readonly IMapper mapper;

        public TaskTrackController(ITaskTrackService taskTracksService, 
            IUserTaskService userTaskService, IMapper mapper)

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs (hunk 4 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs]:
```suggestion
            var searchResult = await taskTracksService.GetAllByUserId(userId);
```

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi.
I did comments. Fix them please.

File: DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/IUserTaskService.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.BusinessLayer/Interfaces/IUserTaskService.cs]:
```suggestion
        Task<IEnumerable<UserTaskModel>> GetAllByUserId(int userId);

File: DIMS-Core/DIMS-Core.BusinessLayer/Models/UserTask/UserTaskModel.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+namespace DIMS_Core.BusinessLayer.Models.UserTask
+{
+    public class UserTaskModel
+    {
+        public int UserTaskId { get; set; }
+        public int TaskId { get; set; }
+        public int UserId { get; set; }
+        public int StateId { get; set; }
+
+        public virtual TaskModel Task { get; set; }
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core.BusinessLayer/Models/UserTask/UserTaskModel.cs:12]: public TaskModel Task { get; set; }

**Review by maximsan — APPROVED** [review]
Looks good for me.

**Review by PushMeNow — APPROVED** [review]
Good job

**Pull request merged**